### PR TITLE
feat(logger): serialize got request error

### PR DESCRIPTION
**Pull request by rarkins**

### CODE REVIEW

File: lib/logger/err-serializer.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ import is from '@sindresorhus/is';
 Error.stackTraceLimit = 20;
 
 interface Err {
+  url?: unknown;
```
Comment (viceice) [inline lib/logger/err-serializer.ts:6]:
```suggestion
```

File: lib/logger/err-serializer.ts [hunk 2]
```diff
@@ -3,6 +3,7 @@ import is from '@sindresorhus/is';
...
   body?: unknown;
   response?: {
     body?: unknown;
```
Comment (viceice) [inline lib/logger/err-serializer.ts:9]:
```suggestion
    body?: unknown;
    url?: unknown;
```

File: lib/logger/err-serializer.ts [hunk 3]
```diff
@@ -30,6 +31,9 @@ export default function errSerializer(err: Err): any {
   if (err.stack) {
     response.stack = err.stack;
   }
+  if (err.url) {
+    response.url = err.url;
+  }
```
Comment (viceice) [inline lib/logger/err-serializer.ts:36]:
```suggestion
  if (err.response?.url) {
    response.url = err.response?.url;
  }
```

**Review by viceice — CHANGES_REQUESTED** [review]

**Comment by viceice** [comment]
https://github.com/renovatebot/renovate/blob/master/lib/util/http/legacy.ts#L29-L33

File: lib/logger/err-serializer.ts (hunk 1 above)
Comment (rarkins) [reply lib/logger/err-serializer.ts:6]: image
?

File: lib/logger/err-serializer.ts [hunk 4]
```diff
@@ -30,6 +31,9 @@ export default function errSerializer(err: Err): any {
   if (err.stack) {
     response.stack = err.stack;
   }
+  if (err.response?.url) {
+    response.url = err.response?.url;
```
Comment (rarkins) [inline lib/logger/err-serializer.ts:35]:
```suggestion
    response.url = err.response.url;
```

**Review by viceice — COMMENTED** [review]
L20: response need to get a explicit type like `Record<string, unknown>`

File: lib/logger/err-serializer.ts [hunk 5]
```diff
@@ -16,7 +17,7 @@ interface Err {
 }
 
 export default function errSerializer(err: Err): any {
-  const response = {
+  const response: Record<string, unknown> = {
     ...err,
   };
   if (err.body) {
```
Comment (viceice) [inline lib/logger/err-serializer.ts:23]: Wonder if we need this? as we do a flat copy above anyways. 🤔

**Comment by rarkins** [comment]
I just wanted the URL to be printed by default when we log `err`. I think this is expanding things much further?

**Comment by viceice** [comment]
we need to make some more changes

**Comment by viceice** [comment]
@rarkins  What do you think about the current changes?

**Review by viceice — APPROVED** [review]

**Pull request merged**